src/main/java/fr/esgi/al5_2/Tayarim/controllers/AdministrateurController.java

```diff
@@ -0,0 +1,173 @@
+package fr.esgi.al5_2.Tayarim.controllers;
+
+import fr.esgi.al5_2.Tayarim.dto.proprietaire.*;
+import fr.esgi.al5_2.Tayarim.entities.Administrateur;
+import fr.esgi.al5_2.Tayarim.exceptions.*;
+import fr.esgi.al5_2.Tayarim.services.AdministrateurService;
+import fr.esgi.al5_2.Tayarim.services.AuthService;
+import fr.esgi.al5_2.Tayarim.services.ProprietaireService;
+import jakarta.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/admin")
+public class AdministrateurController {
+
+  private final AdministrateurService administrateurService;
+  private final AuthService authService;
+
+  public AdministrateurController(AdministrateurService administrateurService,
+      AuthService authService) {
+    this.administrateurService = administrateurService;
+    this.authService = authService;
+  }
+
+    /*
+    @PostMapping("")
+    public ResponseEntity<AdministrateurDTO> creerAdministrateur(@Valid @RequestBody AdministrateurCreationDTO administrateurCreationDTO) {
+        return new ResponseEntity<>(
+            administrateurService.creerAdministrateur(administrateurCreationDTO),
+            HttpStatus.CREATED);
+    }*/
+
+  @GetMapping("")
+  public ResponseEntity<List<AdministrateurDTO>> getAdministrateur(
+      @RequestHeader("Authorization") String authHeader) {
+    authService.verifyToken(getTokenFromHeader(authHeader), true);
+
+    return new ResponseEntity<>(
+        administrateurService.getAdministrateur(),
+        HttpStatus.OK
+    );
+  }
+
+  @GetMapping("/{id}")
+  public ResponseEntity<AdministrateurDTO> getAdministrateur(
+      @RequestHeader("Authorization") String authHeader, @PathVariable Long id) {
+
+    authService.verifyToken(getTokenFromHeader(authHeader), true);
+
+    return new ResponseEntity<>(
+        administrateurService.getAdministrateurById(id),
+        HttpStatus.OK
+    );
+  }
+
+  @PutMapping("/{id}")
+  public ResponseEntity<AdministrateurDTO> updateAdministrateur(
+      @RequestHeader("Authorization") String authHeader, @PathVariable Long id,
+      @RequestBody AdministrateurUpdateDTO administrateurUpdateDTO) {
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.naming.AbbreviationAsWordInNameCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Abbreviation in name 'administrateurUpdateDTO' must contain no more than '1' consecutive capital letters.